library/std/src/sys/pal/unix/thread.rs

```diff
@@ -424,8 +424,8 @@ pub fn available_parallelism() -> io::Result<NonZero<usize>> {
                     if !set.is_null() {
                         let mut count: usize = 0;
                         if libc::pthread_getaffinity_np(libc::pthread_self(), libc::_cpuset_size(set), set) == 0 {
-                            for i in 0..u64::MAX {
-                                match libc::_cpuset_isset(i, set) {
+                            for i in 0..u32::MAX {
+                                match libc::_cpuset_isset(i.into(), set) {
```

Comment:
The argument is an unsigned long, so why is this not `c_ulong::MAX`, exactly?

Reply:
Well, because I know too little rust. :)  And perhaps because I follow too closely in the footsteps already trodden, not wanting to deviate too far.
It's not as if the loop won't terminate before any of those values are reached, so in practical terms the only difference would be the ability to get rid of the `.into()`.
Or ... ideally one should be able to use `cpuid_t::MAX`, or (a compilation attempt later) `libc::cpuid_t::MAX`?

Reply:
agreed, realistically it will never matter!

yeah, I mostly just want the alias to be "self-documenting". in this case it works because the alias means the type so the alias does indeed resolve to a type with an associated constant.